### DISCUSSION TITLE
Derive input field from another input field

### DISCUSSION THREAD

**Discussion by ChrisB151** [Budibase/budibase#3473] [Help] [answered]
Hi all,

Looking for any ideas on automating / simplifying input.  I am using the internal database for this project.  I have a form that allows users to create a new entry.  They have to enter a size in both metric and imperial measurements.  I am looking for a way to allow the user to only enter one of the two values and the other be automatically calculated.

E.g. User enters value in mm, value in inches is automatically calculated or User enters value in inches, value in mm is automatically calculated

I have tried putting a condition on one of the fields to update the default value but I was not able to get this to work.  I also looked at doing the same with a button but could not see a means of getting a button to do this simple calculation.

Any ideas?

**Comment by mjashanks** [comment, marked as answer]
Hi Chris, this is possible. I've created a video below that shows my setup. Steps:

1. I have an `Items` table with columns
- `Name` : `text`
- `Width (mm)` : `number`
- `Width (inches)`: `number`
- `Units` : `Options` ("Millimeters" and "Inches")
2. In my `new` form, I have a `Width` field - you have to free type into the `Field` setting. Note that it's not `Width (mm)` or `Width (inches)`
3. In my Save button, for the `Save Row` action, I override the `Width (mm)` and `Width (inches)` fields, with a value returned by javascript. My `Width (mm)` javascript looks like...

```javascript
if ($("Form.Fields.Units") === 'Millimeters')
	return $("Form.Fields.Width")
else 
	return $("Form.Fields.Width") * 0.0393701
```

I.e. depending on the selected units, I do a conversion on the `Width` field. Hope that makes sense.

https://user-images.githubusercontent.com/3524181/142875230-201a0ee0-5635-405c-af1e-cee3ab357480.mp4

**Reply by ChrisB151** [reply]
That's brilliant, exactly what I was after, thank so much for taking the time to make a video of it too. 

Really appreciated.

**Reply by CovantaDataManagement** [reply]
This works like a charm.  Thanks a million!
